DEVELOPING_TIPS.md

```diff
@@ -0,0 +1,52 @@
+# Cycloid Developer Tips
+
+This file gives some tips to how to test, change or upgrade cli, for cycloid developers.
+
+The cli pipelin is available in the [cycloid-stacks](https://github.com/cycloidio/cycloid-stacks/tree/stacks/cycloid-cli) and the prod and staging cli can be checked [here](https://console.cycloid.io/organizations/cycloid/projects/cycloid-cli).
+
+
+## Update ci client 
+
+using swagger stored in https://docs.cycloid.io/api/swagger.yml
+
+To update the cli client you can either use either:
+
+- the **latest swagger version stored in the the [cycloid docs](https://docs.cycloid.io/api/swagger.yml)**. For that, you just need to run ```make generate-client```, which will run a docker local swagger,generate the new client version and then commit it to the cli git repo where it will be tested in the cli pipeline. </br>
+**Note!** After running the command you should validate that the e2e testing has been succefull and then if no issue has been found, you should edit the CHANGELOG.md to add the new version of swagger.
+
+- a **copied swagger file**, if you wish to use another version of the swagger file, you can create a folder called /gen-swagger and copy it inside. Once that's finished you can run ```make generate-local-client``` that will create the new client version files. After, you should change the client/version file with the corresponding BE version.</br> 
+
+
+## CI local testing 
+### Requirements
+
+To perform local test you need:
+- aws installed locally ([official doc](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html)), required to connect to the cycloid docker images stored in a ECR
+- [backend repo local clone](https://github.com/cycloidio/youdeploy-http-api), required to launch local be
+
+### Procedure
+
+In order to automate the procedure has much as possible we tried to create some Makefile targets to help making the testing easier and faster, but it still requires some manual actions.
+
+To launch the e2e tests, locally you should:
+
+1. Connect to cycloid's ECR, by typing ```make ecr-connect```, it will configure aws cli with the cycloid credentials retrieved from vault and login to the ECR
+
+2. Clone BE git repository and change the variable ```LOCAL_BE_GIT_PATH``` in the makefile acordingly
```

Comment:
Is it really a variableto change in makefile or just an env variable to export ?

Reply:
It is a variable set on the Makefile